mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -168,6 +179,7 @@ private[regression] trait GeneralizedLinearRegressionBase extends PredictorParam
     }
 
     val newSchema = super.validateAndTransformSchema(schema, fitting, featuresDataType)
+    if (isSetOffsetCol(this)) SchemaUtils.checkNumericType(schema, $(offsetCol))
```

Comment:
@sethah This is a great point. The new commit does allow offset to be missing when making predictions. I now check validity of offset only when it's set and available in the prediction set. Otherwise, set offset to be zero. Thanks for catching this and contributing the test.